src/main/java/com/blobExample/client/ClientResource.java

```diff
@@ -0,0 +1,130 @@
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+
+@Path("/displayMessage")
+@Produces(MediaType.APPLICATION_JSON)
+public class ClientResource {
+
+    private final String template;
+    private final String defaultName;
+    private final AtomicLong counter;
+    private javax.ws.rs.client.Client jerseyClient;
+    private BlobStore blobStore;
+    private static final Logger LOGGER = LoggerFactory.getLogger(ClientResource.class);
+
+    public ClientResource(String template, String defaultName, Client jerseyClient, BlobStore blobStore) {
+        this.template = template;
+        this.defaultName = defaultName;
+        counter = new AtomicLong();
+        this.jerseyClient = jerseyClient;
+        this.blobStore = blobStore;
+    }
+
+    public Client getJerseyClient() {
+        return jerseyClient;
+    }
+
+    @GET
+    @Timed
+    @Consumes(MediaType.APPLICATION_JSON)
+    public ClientResponse getMessageFromServer(@QueryParam("name") String name) {
+
+        final String clientName = name != null ? name : defaultName;
+        final String message = String.format(template, clientName);
+        final ClientRequest clientRequest = new ClientRequest(clientName, message);
+
+        BlobContext blobContext = null;
+        if (blobStore != null) {
+            blobContext = new SimpleBlobContext("ServerService", "getMessageFromServer");
+
+            Blobs requestBlob = createBlob(createBlobFactory(), blobContext);
+
+            if (requestBlob != null) {
+                Map<String, String> requestBlobMetadata = new HashMap<>();
+                requestBlobMetadata.put("name", name);
+
+                writeBlob(requestBlob, clientRequest, requestBlobMetadata, BlobType.REQUEST);
+            }
+        }
+
+        WebTarget webTarget = jerseyClient.target("http://localhost:9090/hi");
+        Invocation.Builder invocationBuilder = webTarget.request(MediaType.APPLICATION_JSON_TYPE);
+        Response response = invocationBuilder.post(
+                Entity.entity(
+                        clientRequest,
+                        MediaType.APPLICATION_JSON_TYPE
+                )
+        );
+        ServerResponse serverResponse = response.readEntity(ServerResponse.class);
+
+        if (blobStore != null && blobContext != null) {
+            Blobs responseBlob = createBlob(createBlobFactory(), blobContext);
+
+            if (responseBlob != null) {
+                Map<String, String> responseBlobMetadata = new HashMap<>();
+                responseBlobMetadata.put("name", serverResponse.getServerName());
+
+                writeBlob(responseBlob, serverResponse, responseBlobMetadata, BlobType.RESPONSE);
+            }
+        }
+
+        return new ClientResponse(counter.incrementAndGet(), message, serverResponse);
+    }
+
+    private void writeBlob(Blobs blob, Object data, Map<String, String> blobMetadata, BlobType blobType) {
+        blob.write(blobType,
```

Comment:
Error handling is done by the blob-core so something like this will do
```blob.write(blobType,
                ContentType.JSON,
                o -> objectMapper.writeValueAsBytes(data),
                m -> blobMetadata.forEach((key, value) -> m.add(key, value))
        );```

Reply:
How will the `o` stream contain the data when we haven't written that in it.

`
blob.write(blobType,
                ContentType.JSON,
                (outputStream) -> {
                    ObjectMapper objectMapper = new ObjectMapper();
                    try {
                        outputStream.write(objectMapper.writeValueAsBytes(data));
                    } catch (JsonProcessingException e) {
                        e.printStackTrace();
                    } catch (IOException e) {
                        e.printStackTrace();
                    }
                },
                m -> blobMetadata.forEach((key, value) -> m.add(key, value))
        );
`

This is what I have tried and is working.

Reply:
Made the changes. Please view the latest commit.